core/src/web/camera.ts

```diff
@@ -16,29 +18,109 @@ export class CameraPluginWeb extends WebPlugin implements CameraPlugin {
+          try {
+            await cameraModal.componentOnReady();
+            cameraModal.addEventListener('onPhoto', async (e: any) => {
+              const photo = e.detail;
+
+              if (photo === null) {
+                reject('User cancelled photos app');
+              } else if (photo instanceof Error) {
+                reject(photo.message);
+              } else {
+                resolve(await this._getCameraPhoto(photo, options));
+              }
+
+              cameraModal.dismiss();
+              document.body.removeChild(cameraModal);
+            });
+
+            cameraModal.present();
+          } catch (e) {
+            this.fileInputExperience(options, resolve, reject);
+          }
         } else {
-          resolve(await this._getCameraPhoto(photo, options));
+          console.error(`Unable to load PWA Element 'pwa-camera-modal'. See the docs: https://capacitorjs.com/docs/pwa-elements.`);
+          this.fileInputExperience(options, resolve, reject);
         }
+      } else {
+        this.fileInputExperience(options, resolve, reject);
+      }
+    });
+  }
+
+  private fileInputExperience(options: CameraOptions, resolve: any, reject: any) {
+    let input = document.querySelector('#_capacitor-camera-input') as HTMLInputElement;
+
+    const cleanup = () => {
+      input.parentNode && input.parentNode.removeChild(input);
+    };
+
+    if (!input) {
+      input = document.createElement('input') as HTMLInputElement;
+      input.id = '_capacitor-camera-input';
+      input.type = 'file';
+      document.body.appendChild(input);
+    }
+
+    input.accept = 'image/*';
+    (input as any).capture = true;
 
-        cameraModal.dismiss();
-        document.body.removeChild(cameraModal);
+    if (options.source === CameraSource.Photos || options.source === CameraSource.Prompt) {
+      input.removeAttribute('capture');
+    } else if (options.direction === CameraDirection.Front) {
+      (input as any).capture = 'user';
+    } else if (options.direction === CameraDirection.Rear) {
+      (input as any).capture = 'environment';
+    }
+
+    input.addEventListener('change', (_e: any) => {
+      const file = input.files[0];
+      let format = 'jpeg';
+      if (file.type === 'image/png') {
+        format = 'png';
+      }
+      if (file.type === 'image/gif') {
+        format = 'gif';
+      }
+
+      const reader = new FileReader();
+      reader.addEventListener('load', () => {
+        if (options.resultType === CameraResultType.DataUrl) {
+          resolve({
+            dataUrl: reader.result,
+            format
+          } as CameraPhoto);
+        } else if (options.resultType === CameraResultType.Base64) {
+          const b64 = (reader.result as string).split(',')[1];
+          resolve({
+            base64String: b64,
+            format
+          } as CameraPhoto);
+        } else {
+          reject('Unsupported result type for this platform');
+        }
```

Comment:
```suggestion
        }
```
No need for this else as the reader.readAsDataURL(file); (line 113) is only called for DataUrl and Base64